### PR TITLE
Add a pipeline for tensor.pad.  

### PR DESCRIPTION
**Pull request by pifon2a**
The pipeline fails during `transform.iree.bufferize`.
    
    ```
    requires exactly a single HAL::ExecutableOp or HAL::ExecutableVariantOp target op.
    │ 57     : (!pdl.operation) -> !pdl.operation
    | hal.executable.variant public @cuda_nvptx_fb, target = #executable_target_cuda_nvptx_fb
    ```

### CODE REVIEW

File: transform_dialect/examples/cuda/pad_via_pack_pipeline_spec.mlir [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    : (!pdl.operation) -> !pdl.operation
+  %func_v_2 = transform.vector.lower_masked_transfers %func_v
+    : (!pdl.operation) -> !pdl.operation
+  transform.iree.apply_patterns %func_v_2 { rank_reducing_linalg, rank_reducing_vector }
+    : (!pdl.operation) -> ()
+  %func_v_3 = transform.structured.vectorize %func_v_2
+
+  transform.print %func_v_3 {name = "after vectorization"} : !pdl.operation
+
+  // Step 5. Bufferize and drop HAL descriptor from memref ops.
```
Comment (nicolasvasilache) [inline transform_dialect/examples/cuda/pad_via_pack_pipeline_spec.mlir:94]: Could we just drop this and everything below for now to land and iterate from a common state?

Comment (pifon2a) [reply transform_dialect/examples/cuda/pad_via_pack_pipeline_spec.mlir:103]: done

File: transform_dialect/examples/pack.mlir [hunk 2]
```diff
@@ -0,0 +1,12 @@
+func.func @pack_static(%arg: tensor<999x1999xf32>)
```
[the remark this answers is not in the document]
Comment (pifon2a) [reply transform_dialect/examples/pack.mlir]: done

**Comment by pifon2a** [comment]
@nicolasvasilache Thanks!

**Review by nicolasvasilache — APPROVED** [review]

**Pull request merged**